src/qibolab/instruments/qblox/controller.py

```diff
@@ -41,48 +44,40 @@ def connect(self):
         if self.is_connected:
             return
         try:
-            self.cluster.connect()
-            for name in self.modules:
-                self.modules[name].connect()
+            # Connect cluster
+            QbloxCluster.close_all()
+            self.cluster = QbloxCluster(self.name, self.address)
+            self.cluster.reset()
+            self.cluster.set("reference_source", self._reference_clock)
+            # Connect modules
+            for module in self.modules.values():
+                module.connect(self.cluster)
             self.is_connected = True
+            log.info("QbloxController: all modules connected.")
+            return
```

Comment:
The `return` in this position is redundant (it will terminate anyhow, since there is no instruction after the `try... except` block)